presto-geospatial-toolkit/src/test/java/com/facebook/presto/geospatial/rtree/TestFlatbush.java

```diff
@@ -0,0 +1,281 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.geospatial.rtree;
+
+import com.esri.core.geometry.ogc.OGCGeometry;
+import com.facebook.presto.geospatial.GeometryUtils;
+import com.facebook.presto.geospatial.Rectangle;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestFlatbush
+{
+    private static final Rectangle everything = new Rectangle(
+            Double.NEGATIVE_INFINITY,
+            Double.NEGATIVE_INFINITY,
+            Double.POSITIVE_INFINITY,
+            Double.POSITIVE_INFINITY);
+
+    private static final Comparator<Rectangle> rectangleComparator = Comparator
+            .comparing(Rectangle::getXMin)
+            .thenComparing(Rectangle::getYMin)
+            .thenComparing(Rectangle::getXMax)
+            .thenComparing(Rectangle::getYMax);
+
+    //  2 intersecting polygons: A and B
+    private static final OGCGeometry POLYGON_A = OGCGeometry.fromText("POLYGON ((0 0, -0.5 2.5, 0 5, 2.5 5.5, 5 5, 5.5 2.5, 5 0, 2.5 -0.5, 0 0))");
+    private static final OGCGeometry POLYGON_B = OGCGeometry.fromText("POLYGON ((4 4, 3.5 7, 4 10, 7 10.5, 10 10, 10.5 7, 10 4, 7 3.5, 4 4))");
+
+    // A set of points: X in A, Y in A and B, Z in B, W outside of A and B
+    private static final OGCGeometry POINT_X = TestUtils.makePoint(1, 1);
+    private static final OGCGeometry POINT_Y = TestUtils.makePoint(4.5, 4.5);
+    private static final OGCGeometry POINT_Z = TestUtils.makePoint(6, 6);
+    private static final OGCGeometry POINT_W = TestUtils.makePoint(20, 20);
+
+    @Test
+    public void testEmptyFlatbush()
+    {
+        Rectangle[] items = {};
+        Flatbush rtree = new Flatbush<Rectangle>(items);
+        assertEquals(rtree.findIntersections(everything).size(), 0);
+    }
+
+    @Test
+    public void testSingletonFlatbush()
+    {
+        List<Rectangle> items = new ArrayList();
```

Comment:
- use ImmutableList.of(...)
- specify parameter for Flatbush: `Flatbush<Rectangle> rtree = ...`
- inline hit and miss variables
- ditto testSingletonFlatbushXY

```
        List<Rectangle> items = ImmutableList.of(new Rectangle(0, 0, 1, 1));
        Flatbush<Rectangle> rtree = new Flatbush<>(items.toArray(new Rectangle[] {}));
        assertEquals(findIntersections(rtree, EVERYTHING), items);
        // hit
        assertEquals(findIntersections(rtree, new Rectangle(1, 1, 2, 2)), items);
        // miss
        assertEquals(findIntersections(rtree, new Rectangle(-1, -1, -0.1, -0.1)).size(), 0);
```